gitlab/tests/conftest.py

```diff
@@ -71,18 +72,53 @@ def dd_environment():
 
 @pytest.fixture()
 def mock_data():
-    f_name = os.path.join(os.path.dirname(__file__), 'fixtures', 'metrics.txt')
-    with open(f_name, 'r') as f:
-        text_data = f.read()
     with mock.patch(
         'requests.get',
-        return_value=mock.MagicMock(
-            status_code=200, iter_lines=lambda **kwargs: text_data.split("\n"), headers={'Content-Type': "text/plain"}
-        ),
+        side_effect=mocked_requests_get,
     ):
         yield
 
 
+def mocked_requests_get(*args, **kwargs):
+    if args[0].startswith("http://{}:{}/-/readiness".format(HOST, GITLAB_LOCAL_PORT)):
+        f_name = os.path.join(os.path.dirname(__file__), 'fixtures', 'readiness_check.json')
```

Comment:
Does it make sense here to use our `mock_http_response` fixture?

Reply:
Not sure how to use it in my case since this function is a side_effect 😕

Reply:
gotcha. I'm too beat to investigate :)